core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -111,6 +116,9 @@
   /** The manager of safe mode state. */
   protected final SafeModeManager mSafeModeManager;
 
+  /** The worker configuration checker. */
+  private final ServerConfigurationChecker mWorkerConfigChecker;
```

Comment:
I think it would be better to init configuration in startMasters and in initializing AlluxioMasterProcess.  Are there a need to init it again in stopMasters?

I am not quite sure where should I init configuration so that it can be init when a master become primary.
```
  public ServerConfigurationChecker() {
    init();
  }

  public void init() {
    mConfMap = new HashMap<>();
    mLostConfMap = new HashMap<>();
  }
```
Thanks!

Reply:
If there is a `reset` or `init` for the `ServerConfigurationChecker`, then in `stopMasters` you can just reset the state there.

Reply:
Add a init function in ServerConfigurationChecker
```
  /**
   * Initializes the default {@link ServerConfigurationChecker}.
   */
  public synchronized void init() {
    mConfMap = new HashMap<>();
    mLostNodes = new HashSet<>();
  }
```

Add call this function in AlluxioMasterProcess.stopMasters()
```
  protected void stopMasters() {
    try {
      mWorkerConfigChecker.init();
      mRegistry.stop();
    } catch (IOException e) {
      throw Throwables.propagate(e);
    }
  }
```
Thanks!

Reply:
Actually I agree with Lu that it's better to call `init` during start instead of during stop. It's better to initialize closer to when we need it so that there's less time for something to interfere. When we call stop() there could still be outstanding RPC threads trying to register workers and modifying the server config checker